src/services/snapshots.actions.ts

```diff
@@ -736,34 +881,76 @@ export function updateV4GroupUrls(snapshot: NormalizedSnapshot): void {
   }
 }
 
+export async function getStoredSnapshots() {
+  let stored
+  try {
+    stored = await browser.storage.local.get<Stored>('snapshots')
+  } catch (err) {
+    return Logs.err('Snapshots.vue: recalcSizes: Cannot get snapshots', err)
+  }
+  return stored.snapshots
+}
+
+export async function prepareExport(snapshot: Snapshot | SnapshotState) {
+  // const parsed = parseSnapshot([snapshot], 0)
+  // console.log({parsed});
+
+  // if(!parsed)return
+  const { id, time, containers, sidebar, tabs } = snapshot //parsed
+  const normSnapshot = { id, time, containers, sidebar, tabs }
```

Comment:
`const normSnapshot` is not `NormalizedSnapshot` and that may cause issues in `convertToMarkdown` fn. It's better to use `NormalizedSnapshot` type instead of `Snapshot | SnapshotState` and update:
- `exportSnapshot(...)` to `exportSnapshot(snapshot: NormalizedSnapshot)`
- in `createSnapshot(...)` fn, pass a `NormalizedSnapshot`- compatible type instead of the `Snapshot` to the `exportSnapshot()`:
```ts
  if (Settings.state.snapAutoExport) {
    void exportSnapshot(currentSnapshot)
  }
```
with
```ts
  if (Settings.state.snapAutoExport) {
    exportSnapshot({
      id: currentSnapshot.id,
      time: currentSnapshot.time,
      containers: stored.containers,
      sidebar: stored.sidebar,
      tabs,
    })
  }
```

Reply:
https://github.com/onezoomin/sidebery/commit/24016a2b19357a8b23b273cd7090b2b45d87e26a